frontend/src/metabase/sharing/components/EmailAttachmentPicker.jsx

```diff
@@ -0,0 +1,206 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+
+import ButtonGroup from "metabase/components/ButtonGroup";
+import CheckBox from "metabase/components/CheckBox";
+import StackedCheckBox from "metabase/components/StackedCheckBox";
+import Toggle from "metabase/components/Toggle";
+
+export default class EmailAttachmentPicker extends Component {
+  state = {
+    attachmentType: null,
+    selectedCardIds: new Set(),
+  };
+
+  static propTypes = {
+    pulse: PropTypes.object.isRequired,
+    setPulse: PropTypes.func.isRequired,
+  };
+
+  DEFAULT_ATTACHMENT_TYPE = "csv";
+
+  updateStateFromCards = () => {
+    const { cards } = this.props;
+    const { attachmentType } = this.state;
+    const selectedCards = cards.filter(card => {
+      return card.include_csv || card.include_xls;
+    });
+
+    if (!attachmentType && selectedCards.length > 0) {
+      this.setState({
+        attachmentType: this.attachmentTypeFor(selectedCards),
+        selectedCardIds: new Set(
+          selectedCards.map(card => {
```

Comment:
Also syntax stuff that's not worth changing, but I see a few places where it'd be more concise to use a bracket-less function. here it'd be: `selectedCards.map(card => card.id)`

It's mostly just nice to avoid extra lines.

Reply:
:+1: 

I've been gunshy about syntactic sugar that I've misused recently, but it's definitely time to get over that